### PR TITLE
Arm backend: Add STABLE softmax decomposition config for Ethos-U

### PR DESCRIPTION
**Pull request by JakeStevens**
Summary:
Current behavior for U55 defaults to UNSTABLE. It seems like this is because mask is not supported on U55, but there does not seem to be an inherent need to couple this. Defaulting to UNSTABLE negatively impacts quantization performance.

This PR adds a new `STABLE` option to `SoftmaxDecompositionConfig` that provides numerically stable softmax decomposition without masked fill decomposition.

The three softmax configs now behave as follows:
- `MASKED`: Stable softmax (with amax subtraction) + masked fill decomposition
- `UNSTABLE`: Unstable softmax (no amax subtraction), no masked fill decomposition
- `STABLE`: Stable softmax (with amax subtraction), no masked fill decomposition

For Ethos-U55 targets, `disable_masked_softmax()` now sets the config to `STABLE` instead of `UNSTABLE`, providing numerically stable softmax while avoiding masked fill decomposition which is not needed for these targets.

Differential Revision: D92058235

cc @freddan80 @per @zingo @oscarandersson8218 @digantdesai

### CODE REVIEW

**Comment by JakeStevens** [comment]
I see amax is not supported on u55. Is just max supported? Is there really no support for a stable softmax?

**Comment by AdrianLundell** [comment]
Hi, thanks for your PR. Amax support was added for Ethos-U55 just recently with the latest Vela update, so it is just a matter of enabling it in our backend and then the stable softmax decomposition should be available also for U55. I'll start working on it.

**Comment by JakeStevens** [comment]
> Hi, thanks for your PR. Amax support was added for Ethos-U55 just recently with the latest Vela update, so it is just a matter of enabling it in our backend and then the stable softmax decomposition should be available also for U55. I'll start working on it.

Great, I have created:

https://github.com/pytorch/executorch/issues/17211

to help track.

I will wait on this PR until then.

Thanks!

**Comment by JakeStevens** [comment]
@AdrianLundell  @zingo  what do you think?

I've had teammates confirm this solved issues with QAT and the unstable version

**Review by AdrianLundell — APPROVED** [review]
I like it, thanks for creating tests as well.
